### PR TITLE
Add doc examples for `range::RangeArgument::{start,end}`.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libcollections/range.rs [hunk 1]
```diff
@@ -23,13 +23,45 @@ pub trait RangeArgument<T> {
...
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(collections)]
+    /// #![feature(collections_range)]
+    ///
+    /// extern crate collections;
+    ///
+    /// # fn main() {
```
Comment (GuillaumeGomez) [inline src/libcollections/range.rs]: It doesn't work without putting `main` function?

Comment (frewsxcv) [inline src/libcollections/range.rs]: `extern crate` and feature flags here shouldn't be within `main`, right?

Comment (GuillaumeGomez) [inline src/libcollections/range.rs]: It shouldn't but it works and it is not (the generation is "clever" *normally*).

Comment (frewsxcv) [inline src/libcollections/range.rs]: It works because rustdoc knows what lines to extract and put them outside `main`?

Comment (GuillaumeGomez) [inline src/libcollections/range.rs]: Take a look [here](https://github.com/rust-lang/rust/blob/master/src/librustdoc/test.rs#L350-L366).

Comment (frewsxcv) [inline src/libcollections/range.rs]: Thanks for the info! Fixed in the latest force push.

**Comment by frewsxcv** [comment]
@GuillaumeGomez Check out the latest commit. It seems to fix the build. Have any idea why? I don't have too much time right now, so I might just squash the commits.

**Comment by GuillaumeGomez** [comment]
No idea. Certainly because of the "extern crate". However, I'm a bit surprised there is an external import for collections.

**Comment by GuillaumeGomez** [comment]
The situation of this trait is really strange, however it's used in `std` stable code so let's merge it. Thanks!

@bors: r+ rollup

**Comment by alexcrichton** [comment]
@bors: retry

* https://github.com/rust-lang/rust/pull/35187

**Pull request merged**